benchcab/benchcab.py

```diff
@@ -102,21 +93,41 @@ def _validate_environment(self, project: str, modules: list):
                 )
                 sys.exit(1)
 
-    def _initialise_tasks(self) -> list[Task]:
+    @functools.cache
```

Comment:
After looking at this again, `functools.cache` might be a bit overkill. It also looks like functions that take in a dictionary cannot be memoized with `functools.cache`. We can achieve caching here by setting an initial value for `self._config` to `None` and only read the config if `self._config == None`, else return the value of `self._config`.